core/trino-main/src/test/java/io/trino/sql/TestExpressionInterpreter.java

```diff
@@ -1498,6 +1498,14 @@ public void testRowSubscript()
     {
         assertOptimizedEquals("ROW(1, 'a', true)[3]", "true");
         assertOptimizedEquals("ROW(1, 'a', ROW(2, 'b', ROW(3, 'c')))[3][3][2]", "'c'");
+
+        assertOptimizedEquals("ROW(1, null)[2]", "null");
+        assertOptimizedEquals("ROW(0 / 0, 1)[1]", "ROW(0 / 0, 1)[1]");
+        assertOptimizedEquals("ROW(0 / 0, 1)[2]", "1");
```

Comment:
This is an interesting case. Arguably, this should keep the original expression intact, as the optimized version has different semantics -- the arguments to ROW are expected to be evaluated eagerly, so the expression should fail during evaluation regardless of what optimizations get applied.

Reply:
OK. I'll remove the changes related to Row. Are there any other over-optimizations?